wfsim/core.py

```diff
@@ -608,9 +646,10 @@ def _luminescence_timings_simple(n, dG, E0, r, dr, rr, alpha, uE, p, n_electron,
             ne = n_electron[i]
             dt = dr / (alpha * E0[i] * rr)
             dy = E0[i] * rr / uE - 0.8 * p  # arXiv:physics/0702142
+            avgt = np.sum(np.cumsum(dt) * dy) / np.sum(dy)
 
             j = np.argmax(r <= dG[i])
-            t = np.cumsum(dt[j:])
+            t = np.cumsum(dt[j:]) - avgt
```

Comment:
Why do remove the yield-weighted average time? This yields to negative time, is this supposed to happen? Also this average accounts for values outside the physical path of the electron in case of warping (as you take the larges gas gap for r)

Reply:
Yeah, we do need this time shift, unfortunately, this is because the gate time we measure from data (the time shift applied to all S2s) is using the S2 center time. If we don't shift SE by its center time here, all the S2 center time will be reproduced a bit late. We will see Z_reconstructed deeper than Z_instruction.